mismi-s3/main/s3.hs

```diff
@@ -64,6 +67,10 @@ main = do
   forM_ [sigINT, sigTERM, sigQUIT] $ \s ->
     installHandler s (Catch . void . exitImmediately $ ExitFailure 111) Nothing
 
+  -- set number of capabilities
+  numProcs <- liftIO $ getNumProcessors
+  liftIO $ setNumCapabilities numProcs
```

Comment:
Yeah, so this is equivalent to `+RTS -N`

Depending on the program you may want to still allow it to be set by the command line. You could do something fancier, like `getNumCapabilities` and only set it if it's currently `1`, or something like that.

Something else that's worth thinking about are machine with 32 or 64 cores, I don't think GHC scales that well, so assigning that many capabilities is just a waste.